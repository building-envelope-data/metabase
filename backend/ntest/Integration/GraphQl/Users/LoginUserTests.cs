using System.Collections.Generic;
using System.IO;
using System.Threading.Tasks;
using FluentAssertions;
using Snapshooter.NUnit;
using NUnit.Framework;

namespace Metabase.Tests.Integration.GraphQl.Users
{
    [TestFixture]
    public sealed class LoginUserTests
      : UserIntegrationTests
    {
        [Test]
        public async Task ValidDataOfRegisteredAndConfirmedUser_LogsInUser()
        {
            // Arrange
            var email = "john.doe@ise.fraunhofer.de";
            var password = "aaaAAA123$!@";
            await RegisterAndConfirmUser(email: email, password: password).ConfigureAwait(false);
            // Act
            var response = await SuccessfullyQueryGraphQlContentAsString(
                File.ReadAllText("Integration/GraphQl/Users/LoginUser.graphql"),
                variables: new Dictionary<string, object?>
                {
                    ["email"] = email,
                    ["password"] = password,
                    ["rememberMe"] = false
                }
                ).ConfigureAwait(false);
            // Assert
            // TODO assert that cookie was set!
            Snapshot.Match(
                response,
                matchOptions => matchOptions.Assert(fieldOptions =>
                 fieldOptions.Field<string>("data.loginUser.user.id").Should().NotBeNullOrWhiteSpace()
                 )
                );
        }

        // TODO Other cases!
    }
}