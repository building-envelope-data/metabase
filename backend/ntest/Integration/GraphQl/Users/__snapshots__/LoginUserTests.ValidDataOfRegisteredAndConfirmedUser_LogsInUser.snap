{"data":{"loginUser":{"requiresTwoFactor":false,"user":{"id":"VXNlcgpnYmE4NTE0ZGVlZGExNDU1Njk4ZjNmMGNkMTBmNjIwNmQ=","email":null},"errors":null}}}
